contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -568,13 +593,18 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
     function _sendTeleporterMessage(
         TeleporterMessageInput memory messageInput,
         TeleporterMessageReceipt[] memory receipts
-    ) private returns (uint256) {
+    ) private returns (bytes32) {
+        // If the blockchain ID has yet to be initialized, do so now.
+        bytes32 blockchainID_ = _initializeBlockchainID();
+
         // Get the message ID to use for this message.
-        uint256 messageID = _getNextMessageID(messageInput.destinationBlockchainID);
+        uint256 messageNonce_ = messageNonce;
```

Comment:
Is the reason for this local variable just to make the function less error prone for developers? From a correctness standpoint, I think we can use `messageNonce` directly, as long as we finish using it before we increment.

Reply:
It's actually a gas optimization that Louis called out. `messageNonce` is kept in storage, so reading it once into a local variable reduces the total number of storage reads since we use the value more than once.

Reply:
Nice, makes sense